Connectors/Navisworks/Speckle.Connectors.NavisworksShared/Operations/Send/Filters/NavisworksSavedViewsFilter.cs

```diff
@@ -0,0 +1,173 @@
+    // {
+    //   // We check this again as the view settings may have changed in the saved card.
+    //   // If the saved view does not contain visibility overrides, this is effectively everything in the model.
+    //   // This will need to be the documented behaviour.
+    //   throw new SpeckleSendFilterException(
+    //     "Saved view does not contain visibility overrides. This would effectively publish everything in the model."
+    //   );
+    // }
+
+    NavisworksApp.ActiveDocument.SavedViewpoints.CurrentSavedViewpoint = savedView;
+    var models = NavisworksApp.ActiveDocument.Models;
+    NavisworksApp.ActiveDocument.CurrentSelection.Clear();
+
+    foreach (var model in models)
+    {
+      var rootItem = model.RootItem;
+
+      if (!_selectionService.IsVisible(rootItem))
+      {
+        // If the root item is hidden, we skip it and its descendants.
+        continue;
+      }
+
+      objectIds.AddRange(
+        rootItem.Descendants.Where(_selectionService.IsVisible).Select(_selectionService.GetModelItemPath).ToList()
+      );
+    }
+
+    return objectIds;
+  }
+
+  /// <summary>
+  /// Since it is called from constructor, it is re-called whenever UI calls SendBinding.GetSendFilters() on SendFilter dialog.
+  /// Do not change the behavior/scope of this class on send binding unless make sure the behavior is same. Otherwise, we might not be able to update list of saved sets.
+  /// </summary>
+  private void GetSavedViews()
+  {
+    List<NAV.SavedViewpoint> savedViewRecords = [];
+
+    var root = NavisworksApp.ActiveDocument.SavedViewpoints.RootItem;
+
+    CollectSavedViews(root, savedViewRecords);
+
+    Items = savedViewRecords
+      .Select(viewRecord =>
+      {
+        var reference = NavisworksApp.ActiveDocument.SavedViewpoints.CreateReference(viewRecord);
+
+        // If the guid is effectively empty, we can use the saved view's name as a fallback
+        var selectItemId =
+          viewRecord.Guid.ToString() == Guid.Empty.ToString() ? reference.SavedItemId : viewRecord.Guid.ToString();
+        string hierarchicalName = SavedItemHelpers.BuildHierarchicalName(viewRecord, root);
+
+        return new SendFilterSelectItem(selectItemId, hierarchicalName);
+      })
+      .ToList();
+  }
+
+  private static void CollectSavedViews(NAV.SavedItem parentItem, List<NAV.SavedViewpoint> collectedSets)
+  {
+    if (!parentItem.IsGroup)
+    {
+      return;
+    }
+
+    foreach (NAV.SavedItem item in ((NAV.GroupItem)parentItem).Children)
+    {
+      switch (item.IsGroup)
+      {
+        // THIS IS COMMENTED OUT AS IT IS LEGACY DEFENSIVE BEHAVIOUR - DISCUSSION REQUIRED
+        // case false when item is NAV.SavedViewpoint { ContainsVisibilityOverrides: false }:
+        //   // If the saved view does not contain visibility overrides, this is effectively everything in the model.
+        //   // This will need to be the documented behaviour.
+        //   break;
```

Comment:
Willing to discuss either uncommenting or removing this. An opinion expressed in v2 code was that this was valuable, partly because sends were slower then, but also because this class of viewpoint doesn't hide anything from view; it's no different from sending everything.

This PR keeps the legacy position but commented out.

Reply:
same as before